src/script.js

```diff
@@ -0,0 +1,231 @@
+   tasks.sortDown();
+   renderTasks();
+};
+
+sortUpButton.onclick = () => {
+   tasks.sortUp();
+   renderTasks();
+};
+
+function renderTasks() {
+   let lastTaskAdded;
+   while (allTasksContainer.firstChild) {
+      allTasksContainer.removeChild(allTasksContainer.lastChild);
+   }
+   tasks.items.forEach((item) => {
+      const taskContainer = createTaskContainer();
+      const newTask = createNewTask(taskContainer, item);
+      const deleteTask = createDeleteTaskButton(taskContainer);
+      addHoverReveal(taskContainer, deleteTask, newTask);
+      lastTaskAdded = newTask;
+   });
+
+   return lastTaskAdded;
+}
+
+async function addTask() {
+   const taskUserInput = taskInput.value;
+   taskInput.value = "";
+   const allPokemonNames = await pokemonClient.getAllPokemonNames();
+   if (taskUserInput.trim().length === 0) {
+      alert("Please fill in a task");
+
+      return false;
+   } else if (
+      !isNaN(taskUserInput) ||
+      allPokemonNames.includes(taskUserInput)
+   ) {
+      // For single Pokemon entry
+      let pokemonData = await pokemonClient.getPokemon(taskUserInput);
+      let render = false;
+      render = pokemonTasksHandle(pokemonData, taskUserInput, 0);
+
+      return render;
+   } else if (
+      taskUserInput
+         .replace(/\s/g, "")
+         .split(",")
+         .every((elem) => !isNaN(elem) || allPokemonNames.includes(elem))
+   ) {
+      // For multiple Pokemons entry
+      let pokemonIDS = taskUserInput.replace(/\s/g, "").split(","); // "1, 2, 3" => [1,2,3]
+      const pokemonData = await Promise.all(
+         pokemonIDS.map((id) => pokemonClient.getPokemon(id))
+      );
+      let render = false;
+      pokemonData.forEach((pokemon, i) => {
+         render = pokemonTasksHandle(pokemon, pokemonIDS, 1, i) || render; // Render even if one pokemon is not listed in tasks
+      });
+
+      return render;
+   } else {
+      // For regular boring non-pokemon tasks
+      tasks.add(taskUserInput);
+      pendingTasksUpdate("+");
+
+      return true;
+   }
+}
+
+function pokemonTasksHandle(pokemon, pokemonIDS, isMultiplePokemons, i) {
+   let render = false;
+   if (pokemon) {
+      let pokemonName =
+         pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1);
+      const pokemonTypes = pokemonClient.getPokemonTypes(pokemon);
+      const taskToAdd = `Catch ${pokemonName} of type ${pokemonTypes}`;
+      if (tasks.items.includes(taskToAdd)) {
+         alert(
+            `${pokemonName} already exists in your tasks. Please try another Pokemon.`
+         );
+         render = false;
+      } else {
+         addPokemonImage(pokemon);
+         tasks.add(taskToAdd);
+         pendingTasksUpdate("+");
+
+         render = true;
+      }
+   } else {
+      if (isMultiplePokemons) {
+         tasks.add(`Pokemon ID ${pokemonIDS[i]} does not exist`);
+      } else {
+         tasks.add(`Pokemon ID ${pokemonIDS} does not exist`);
+      }
+      pendingTasksUpdate("+");
```

Comment:
now that you have ItemManager, think how you can leverage it to manage the pending tasks status and counter